app/lib/language-selector.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component, PropTypes } from 'react';
+import Select from 'react-select';
+import languages from './languages';
+import { select } from './language-engine';
+
+const propTypes = {
+  props: PropTypes.object.isRequired
+};
+const storedLanguage = JSON.parse(localStorage.getItem('preferred-language'));
+const defaultLanguage = { label: 'English', value: 'en' };
```

Comment:
default language is defined in more than one place, which could make it harder to change.